src/Microsoft.Extensions.Configuration/ConfigurationRoot.cs

```diff
@@ -3,34 +3,88 @@
 
 using System;
 using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Microsoft.Extensions.Primitives;
 
 namespace Microsoft.Extensions.Configuration
 {
-    public class ConfigurationRoot : ConfigurationBase, IConfigurationRoot
+    public class ConfigurationRoot : IConfigurationRoot
     {
+        private IList<IConfigurationProvider> _providers;
+        private ConfigurationReloadToken _reloadToken = new ConfigurationReloadToken();
+
         public ConfigurationRoot(IList<IConfigurationProvider> providers)
-            : base(providers)
         {
             if (providers == null)
             {
                 throw new ArgumentNullException(nameof(providers));
             }
+
+            _providers = providers;
         }
 
-        public override string Path
```

Comment:
Can't say - was just making the implementations follow the interface. If the interfaces are changed in a way that you've described it'd be added back of course, but folks won't be down-casting to these classes anyway so having it or not shouldn't be a material difference...